components/motor/gpio/motor_encoder_test.go

```diff
@@ -4,755 +4,279 @@ import (
 	"context"
 	"sync"
 	"testing"
-	"time"
 
 	"go.viam.com/test"
 	"go.viam.com/utils/testutils"
 
-	"go.viam.com/rdk/components/board"
-	fakeboard "go.viam.com/rdk/components/board/fake"
 	"go.viam.com/rdk/components/encoder"
-	"go.viam.com/rdk/components/encoder/incremental"
-	"go.viam.com/rdk/components/encoder/single"
 	"go.viam.com/rdk/components/motor"
-	fakemotor "go.viam.com/rdk/components/motor/fake"
 	"go.viam.com/rdk/logging"
-	"go.viam.com/rdk/operation"
 	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/testutils/inject"
 )
 
-func nowNanosTest() uint64 {
-	return uint64(time.Now().UnixNano())
-}
+const (
+	encoderName = "encoder"
+	motorName   = "motor"
+)
 
-type fakeDirectionAware struct {
-	m *fakemotor.Motor
+type injectedState struct {
+	mu       sync.Mutex
+	position float64
+	powerPct float64
 }
 
-func (f *fakeDirectionAware) DirectionMoving() int64 {
-	return int64(f.m.Direction())
+var vals = injectedState{
+	position: 0.0,
+	powerPct: 0.0,
```

Comment:
So the tests aren't mutating a global variable, you can have this be created by a function when you need it. 

injectEncoder can also take int the values and lock them.